engine/lib/ast.ml

```diff
@@ -217,6 +217,7 @@ functor
 
     and pat = { p : pat'; span : span; typ : ty }
     and field_pat = { field : global_ident; pat : pat }
+    and cf_kind = BreakOnly | BreakOrReturn
```

Comment:
Can you add a comment here to explain what it is, where it is used and why (e.g. we need that to track what control flow was present in the loop after removing that control flow in favor of simple values representing this control flow)?